clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -549,7 +549,13 @@ private boolean awaitNodeReady(Node node, FindCoordinatorRequest.CoordinatorType
     private void handleProduceResponse(ClientResponse response, Map<TopicPartition, ProducerBatch> batches, long now) {
         RequestHeader requestHeader = response.requestHeader();
         int correlationId = requestHeader.correlationId();
-        if (response.wasDisconnected()) {
+        if (response.wasTimedOut()) {
+            log.trace("Cancelled request with header {} due to node {} being disconnected due to timeout",
```

Comment:
very nit: `due to the last request timed out`? just to distinguish from connection timeout.

Reply:
Changed to:

>Cancelled request with header {} due to the last request to node {} timed out